Add static Loop methods

clue opened the pull request:
This changeset adds static Loop methods. The Loop class now provides all methods that exist on the LoopInterface as static methods:

```php
use React\EventLoop\Loop;

// old (still supported)
Loop::get()->addTimer(1.0, fn () => echo 'Tick');
Loop::get()->run();

// new
Loop::addTimer(1.0, fn () => echo 'Tick');
Loop::run();
```

This is particularly useful when directly interfacing with the event loop in application code. I've also updated the documentation to highlight some use cases where the `Loop::get()` method would still be useful when using dependency injection (DI).

Builds on top of #226

WyriHaximus commented on src/Loop.php:
```diff
@@ -47,4 +47,154 @@ public static function set(LoopInterface $loop)
     }
+
+    /**
+     * [Advanced] Register a listener to be notified when a stream is ready to read.
```
Why are only adding streams marked as advanced? IMHO if we mark those two we should also mark `removeReadStream` && `removeWriteStream` as such. And maybe mention [react/stream](https://reactphp.org/stream/) for easier usage.

clue replied:
Fair point. I've applied the exact same documentation from the `LoopInterface::addReadStream()` method:

https://github.com/reactphp/event-loop/blob/1a709e2847cbd6f61110349792239fde283f4edc/src/LoopInterface.php#L8

Your suggestion sounds perfectly reasonable, perhaps file this as a follow-up PR to also adjust the interface and the README (similar to #110)? :+1:

WyriHaximus replied:
Will do 👍

clue commented:
Rebased on `master` now that #230 is in to trigger a new test run :shipit:

clue commented:
Updated to move deprecation notices out of this PR and split this up into a dedicated PR #231. To ease review, this PR here now only deals with the new static methods.

(Merging either PR will likely cause a merge conflict with the other PR, but I'd rather put the burden of rebasing and fixing this conflict onto myself than the people reviewing this.)
